prisma/schema.prisma

```diff
@@ -0,0 +1,144 @@
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+// Users that are either authors, contributors or participants in slack threads, issues or pull requests
+model User {
+  id             Int     @id @default(autoincrement())
+  githubUsername String?
+  email          String  @unique
+  slackId        String? @unique
+
+  messagesAsked      SlackMessage[]
+  issuesOpened       GithubItem[]
+  itemsResolved      ActionItem[]
+  participatingItems Participant[]
+
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  @@index([id])
+}
+
+model SlackMessage {
+  id   Int      @id @default(autoincrement())
+  text String
+  ts   DateTime
+
+  channelId Int
+  channel   Channel @relation(fields: [channelId], references: [id])
+
+  authorId   Int
+  author     User        @relation(fields: [authorId], references: [id])
+  actionItem ActionItem?
+
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  @@index([id])
+  @@index([authorId])
+  @@index([channelId])
+}
+
+model GithubItem {
+  id         Int            @id @default(autoincrement())
+  number     Int
+  nodeId     String
+  databaseId BigInt
+  state      GithubState
+  type       GithubItemType
+  actionItem ActionItem?
+
+  repositoryId Int
+  repository   Repository @relation(fields: [repositoryId], references: [id])
+
+  authorId Int
+  author   User @relation(fields: [authorId], references: [id])
+
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  @@index([id])
+  @@index([repositoryId])
+  @@index([authorId])
+}
+
+// It can either be a slack message or a github issue / pull request
+model ActionItem {
+  id Int @id @default(autoincrement())
+
+  slackMessageId Int?          @unique
+  slackMessage   SlackMessage? @relation(fields: [slackMessageId], references: [id])
+
+  githubItemId Int?        @unique
+  githubItem   GithubItem? @relation(fields: [githubItemId], references: [id])
+
+  resolvedById Int
+  resolvedBy   User          @relation(fields: [resolvedById], references: [id])
+  participants Participant[]
+
+  project    String
+  status     ActionStatus
+  resolvedAt DateTime?
+
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  @@index([id])
+  @@index([slackMessageId])
+  @@index([githubItemId])
+  @@index([resolvedById])
+}
+
+// Slack channels that are being monitored
+model Channel {
+  id       Int            @id @default(autoincrement())
+  name     String
+  slackId  String
+  messages SlackMessage[]
+
+  @@index([id])
+}
+
+// List of users that are involved in the thread or an issue / pull request
+model Participant {
+  userId Int
+  user   User @relation(fields: [userId], references: [id])
+
+  actionItemId Int
+  actionItem   ActionItem @relation(fields: [actionItemId], references: [id])
+
+  @@id([userId, actionItemId])
+}
+
+model Repository {
+  id    Int    @id @default(autoincrement())
+  name  String
+  owner String
```

Comment:
Should this be a relationship to `User`? similar to `author`

Reply:
it will often be an org account, like `hackclub`, and it will not exist in `User`

Reply:
while `author` is always a user, like `faisalsayed10` or `johndoe`